modules/gateway/synchronize.go

```diff
@@ -24,7 +21,7 @@ func (g *Gateway) Synchronize() (err error) {
 	if err != nil {
 		return
 	}
-	go g.synchronize(peer)
+	g.synchronize(peer)
```

Comment:
hmm. good point. I changed it to make testing easier, but I might have to revert it.